### PR TITLE
Update unit tests for StudentGroupService

### LINKED ISSUE #431: Update unit tests for StudentGroupService

### PR DESCRIPTION
**Pull request by zelya2021**
Break huge tests into smaller ones (by method) and added new tests

### CODE REVIEW

**Review by TetianaDomanska-dev — APPROVED** [review]

**Review by Serwlashev — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs [hunk 1]
```diff
@@ -19,297 +18,362 @@ namespace CharlieBackend.Api.UnitTest
...
+                                  .ReturnsAsync(new List<long>());
+
+            //Act
+            var successResult = await StudentGroupServiceMock().CreateStudentGroupAsync(newStudentGroup);
+            
+            //Assert
+            successResult.Data.Should().NotBeNull();
+            successResult.Data.Name.Should().Be(newStudentGroup.Name);
+            successResult.Data.CourseId.Should().Be(newStudentGroup.CourseId);
+            successResult.Data.StudentIds.Should().BeEquivalentTo(newStudentGroup.StudentIds);
```
Comment (Serhii-Dudar) [inline CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs]: I suppose we don't need test these properties separately. Fleunt assertion provide possibillity to test whole object.

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs [hunk 2]
```diff
@@ -19,297 +18,362 @@ namespace CharlieBackend.Api.UnitTest
...
 
+            //Act
+            var withNotExistCoursIdResult = await StudentGroupServiceMock().UpdateStudentGroupAsync(ExistingStudentGroup().Id, studentGroupWithoutCourseID);
+
+            //Assert
+            withNotExistCoursIdResult.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+        }
+
+        [Fact]
+        public async Task UpdateStudentGroup_StudentGroupWithoutValidStudentIDs_ShouldReturnValidationError()
```
Comment (Serhii-Dudar) [inline CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs]: Better to call this method how UpdateStudentGroup_NotValidStudentIDs_ShouldReturnValidationError

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs [hunk 3]
```diff
@@ -19,297 +18,362 @@ namespace CharlieBackend.Api.UnitTest
...
+        {
+            //Act
+            var nullGroupResult = await StudentGroupServiceMock().UpdateStudentGroupAsync(ExistingStudentGroup().Id, null);
+
+            //Assert
+            nullGroupResult.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+        }
 
+        [Fact]
+        public async Task UpdateStudentGroup_WithNotExistCoursId_ShouldReturnValidationError()
```
Comment (Serhii-Dudar) [inline CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs]: WithNotExistCoursId->NonExistentCourseId

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs (hunk 1 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs]: ok, done

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs (hunk 2 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs]: done

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs (hunk 3 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs]: done

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs [hunk 4]
```diff
@@ -121,9 +121,6 @@ public async Task CreateStudentGroupAsync_StudentGroup_ShouldReturnStudentGroup(
             
             //Assert
             successResult.Data.Should().NotBeNull();
-            successResult.Data.Name.Should().Be(newStudentGroup.Name);
-            successResult.Data.CourseId.Should().Be(newStudentGroup.CourseId);
-            successResult.Data.StudentIds.Should().BeEquivalentTo(newStudentGroup.StudentIds);
```
Comment (Serhii-Dudar) [inline CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs]: I meant to replace that verification to this: successResult.Data.Should().BeEquivalentTo(newStudentGroup);

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs (hunk 2 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs]: got it fixed

File: CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs (hunk 4 above)
Comment (zelya2021) [reply CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs]: got it fixed

**Review by Serhii-Dudar — APPROVED** [review]

**Pull request merged**